source/Nevermore/TableColumnNameResolvers/JsonLastTableColumnNameResolver.cs

```diff
@@ -5,11 +5,11 @@ namespace Nevermore.TableColumnNameResolvers
 {
     public class JsonLastTableColumnNameResolver : ITableColumnNameResolver
     {
-        readonly IReadQueryExecutor queryExecutor;
+        readonly IRelationalStore store;
 
-        public JsonLastTableColumnNameResolver(IReadQueryExecutor queryExecutor)
+        public JsonLastTableColumnNameResolver(IRelationalStore store)
```

Comment:
This needs access to the store rather than the transaction, because it's entirely likely that we'll open a data reader to read one resultset and then need to know the names of the columns in another table. This happens reliably when joining tables but it's relatively easy to provoke elsewhere as well.

By passing the `IRelationalStore` here instead, we can guarantee that the column name resolver will always be able to query for column names, even in the midst of building/running another query.